### ISSUE TITLE
propertyKey of JsonProperty is ostensively undefined

### ISSUE THREAD

**Issue by Kcorab** [dhlab-basel/json2typescript#93]
Firstly the error output:
```
                var decorated = decorator(target, propertyKey, descriptor);
                                ^
Error: Fatal error in JsonConvert. It is not allowed to explicitly pass "undefined" as second parameter in the @JsonProperty decorator.

        Class property: 
                größe

Use "Any" to allow any type. You can import this class from "json2typescript".
```
The problematic class which has the property `größe` is called `Kleidungsstück`.

Please take a look at my code snippet of `kleidungsstück.model.ts`.

```typescript
@JsonObject("Kleidungsstück")
@Entity()
export class Kleidungsstück {
  // ...
  // Doesn't end up in problems
  @JsonProperty('3', Saison)
  @OneToOne(type => Saison, { cascade: true, nullable: true })
  @JoinColumn({ name: "saison" })
  readonly saison = new Saison()

  // End up in problems
  @JsonProperty("12", Größe)
  @OneToOne(type => Größe, größe => größe.kleidungsstück, { cascade: true })
  readonly größe = new Größe()
  // ...
}
```
_json2typescript_ suggests that the type `Größe` is undefined. How can this be?

Now take a look at `größe.model.ts`, please.

```typescript
@JsonObject("Größe")
@Entity()
export class Größe {
  //...
}
```

Do you have some ideas why this happens?

**Comment by andreas-aeschlimann** [comment]
Hi, I cannot tell from this example.

Are you sure that umlauts and the double-s character don't cause a problem?

I have also never worked with numerical properties before, but I guess this should work. What does your initial json look like? Maybe we should create a (minimal) stackblitz example to figure this out.

**Comment by Kcorab** [comment]
After two hours of searching I came to the conclusion that the problem has nothing to do with _json2typescript_.

I thought it was very unlikely that the error was caused by the umlauts and the sharp s (`'ß'`) because everything has the UTF-8 encoding.
Nevertheless I have renamed my `größe.model.ts` to `size.model.ts` and removed all words with umlauts and sharp s.
The result: Same error! But now it becomes strange.

I have **copied** my `size.model.ts` to `größe.model.ts` and restore the removed characters by hand. It works with this copied file (`größe.model.ts`)!
The git compare between the old (corrupt) `größe.model.ts` and the new `größe.model.ts` shows no different! The compiled `.js` files also looked right. Very creepy.

The initially issue is now solved.
